src/status_im/contexts/wallet/swap/utils.cljs

```diff
@@ -30,33 +30,19 @@
     :else
     (i18n/label :t/something-went-wrong-please-try-again-later)))
 
-(defn- first-operable-account
-  [accounts]
-  (->> accounts
-       vals
-       (remove :watch-only?)
-       (filter :operable?)
-       (sort-by :position)
-       first))
-
-(defn wallet-account
-  "Picks the account that's gonna be used for the swap operation. 
-   It's gonna be either the preselected account defined by 
-   `[:wallet :current-viewing-account-address]` in `db`
-   or the first operable account."
+(defn current-viewing-account
   [wallet]
-  (if-let [wallet-address (get wallet :current-viewing-account-address)]
+  (when-let [wallet-address (get wallet :current-viewing-account-address)]
     (-> wallet
         :accounts
         vals
-        (utils/get-account-by-address wallet-address))
-    (-> wallet :accounts first-operable-account)))
+        (utils/get-account-by-address wallet-address))))
```

Comment:
Not related to the PR changes but can't resist. Apologies.

The wallet accounts in app-db are indexed by their address. The above extraction could be:

```clojure
(get-in wallet [:accounts wallet-address])
```

Reply:
Thanks @smohamedjavid for pointing this out. done.